SearchInUnknownSortedArray.java

```diff
@@ -34,3 +34,30 @@ else if (midVal < target){
         return -1;
     }
 }
+
+
+/**
+ * Another approach
+ * Since elements are unique. We can directly find the high as target - reader.get(low)
+ * this will reduce our TC by O(logM) which was used in previous solution for incrementing High by 2
+ */
+
+class Solution {
+    public int search(ArrayReader reader, int target) {        
+        int low = 0; 
+        int high = target - reader.get(low);
+                
+        while (low <= high){        // BS => TC = O(logN)
```

Comment:
But here your N is greater than the previous case , so your log N is bigger than previous solutions log N